src/components/views/context_menus/GenericElementContextMenu.js

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2017 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class GenericElementContextMenu extends React.Component {
```

Comment:
A brief comment on what this does might be useful. It appears to tell a child element about window resizes, but I'm not sure what that has to do with context menus.

Reply:
The position of the contextual menus are set at construction time. If you resize the window while the menu is visible the element that the menu is linked to moves but the menu does not as it is displayed in a DOM element on the document root. By adding a resize listener the menu parent can be triggered to update the position or (in this case) close the menu.